aten/src/ATen/core/dispatch/Dispatcher.cpp

```diff
@@ -217,7 +217,7 @@ void Dispatcher::cleanup(const OperatorHandle& op, const OperatorName& op_name)
   }
 }
 
-RegistrationHandleRAII Dispatcher::registerFallback(DispatchKey dispatchKey, KernelFunction kernel) {
+RegistrationHandleRAII Dispatcher::registerFallback(DispatchKey dispatchKey, KernelFunction kernel, std::string debug) {
```

Comment:
you're not using `debug` here, right? In that case, you could also make it a `const&`.

General recommendation: If a function takes an argument by value or by `const&` should not depend on the call site but only on if the function internally takes ownership (e.g. moves the value further) or not. The API contract (i.e. what the call site needs to know) for by-value and `const&` is the same, it's just a compiler optimization. Both calling conventions look the same at the call site, so by reading you cannot distinguish them. That means the call site should ignore if the function takes an argument by value or as `const&` and just `std::move` it in whenever that is valid (and the type is large enough so moving in general makes sense).

Reply:
This one should morally take it by move; the problem is I'm not actually saving debug information for fallbacks (but I should!) There's a more generic problem which should get fixed at some point, which is we're not doing adequate error checking for fallbacks, c.f. this existing comment:

```
RegistrationHandleRAII Dispatcher::registerFallback(DispatchKey dispatchKey, KernelFunction kernel, std::string debug) {
  // TODO: fallbacks clobber each other completely unsafely, unlike regular
  // kernels
```